src/tests/functional/plugin/cpu/single_layer_tests/reverse_sequence.cpp

```diff
@@ -0,0 +1,221 @@
+        } else {
+            const auto maxSeqLength = dataInputShape.second.front().at(seqAxisIndex);
+            seqLengthsInput = ngraph::builder::makeConstant<float>(seqLengthsPrc, seqLengthsShape.second.front(), {}, true, maxSeqLength);
+        }
+
+        const auto reverse = std::make_shared<ngraph::opset1::ReverseSequence>(paramsIn.front(), seqLengthsInput, batchAxisIndex, seqAxisIndex);
+        const ngraph::ResultVector results{std::make_shared<ngraph::opset1::Result>(reverse)};
+        function = std::make_shared<ngraph::Function>(results, paramsIn, "ReverseSequence");
+    }
+
+    void generate_inputs(const std::vector<ov::Shape>& targetInputStaticShapes) override {
+        inputs.clear();
+        const auto& funcInputs = function->inputs();
+
+        const auto dataInputTensor =
+            ov::test::utils::create_and_fill_tensor(funcInputs[0].get_element_type(),
+                                                    targetInputStaticShapes[0]);
+        inputs.insert({funcInputs[0].get_node_shared_ptr(), dataInputTensor});
+
+        if (funcInputs.size() != 1) {
+            const auto maxSeqLength = targetInputStaticShapes.front().at(m_seqAxisIndex);
+            const auto seqLengthsTensor =
+                ov::test::utils::create_and_fill_tensor(funcInputs[1].get_element_type(),
+                                                        targetInputStaticShapes[1],
+                                                        maxSeqLength, 1);
+            inputs.insert({funcInputs[1].get_node_shared_ptr(), seqLengthsTensor});
+        }
+    }
+
+private:
+    int64_t m_seqAxisIndex;
+};
+
+TEST_P(ReverseSequenceLayerCPUTest, CompareWithRefs) {
+    run();
+}
+
+namespace {
+
+const std::vector<InferenceEngine::Precision> netPrecisions = {
+        InferenceEngine::Precision::FP32,
+        InferenceEngine::Precision::I32
+};
+
+const int64_t batchAxisIndex = 0L;
+
+const std::vector<int64_t> seqAxisIndices = {1L, 2L};
+
+const std::vector<InputShape> dataInputStaticShapes3D = {{{}, {{3, 10, 20}}}, {{}, {{3, 12, 48}}}};
+
+const std::vector<InputShape> dataInputStaticShapes4D = {{{}, {{3, 7, 10, 20}}}, {{}, {{3, 12, 5, 4}}}};
+
+const std::vector<InputShape> dataInputStaticShapes5D = {{{}, {{3, 12, 7, 10, 2}}}, {{}, {{3, 3, 12, 1, 40}}}};
+
+const std::vector<InputShape> seqLengthsStaticShapes = {{{}, {{3}}}};
+
+const std::vector<InputShape> dataInputDynamicShapes3D =
+    {{{7, -1, {5, 10}}, {{7, 20, 8}, {7, 15, 10}}}, {{-1, -1, -1}, {{7, 4, 1}, {7, 42, 70}}}};
+
+const std::vector<InputShape> dataInputDynamicShapes4D =
+    {{{7, -1, {5, 10}, -1}, {{7, 20, 8, 4}, {7, 2, 7, 50}}}, {{-1, -1, -1, -1}, {{7, 15, 1, 100}, {7, 4, 10, 12}}}};
+
+const std::vector<InputShape> dataInputDynamicShapes5D =
+    {{{7, -1, {5, 10}, -1, {2, 14}}, {{7, 3, 8, 20, 9}, {7, 12, 10, 3, 2}}},
+    {{-1, -1, -1, -1, -1}, {{7, 15, 15, 10, 3}, {7, 4, 100, 90, 5}}}};
+
+const std::vector<InputShape> seqLengthsDynamicShapes = {{{-1}, {{7}}}};
```

Comment:
Now that secondaryInputTypes is always ngraph::helpers::InputLayerType::PARAMETER for dynamic shape, I think we should also cover the test cases, where seqLengthsDynamicShapes change dynamicly as well as the batch dimension of dataInputDynamicShapes*D, e.g.
const std::vector<InputShape> dataInputDynamicShapes5D =
    {...,
    {{-1, -1, -1, -1, -1}, {{7, 15, 15, 10, 3}, {10, 4, 100, 90, 5}}}};
const std::vector<InputShape> seqLengthsDynamicShapes = {{{-1}, {{7}, {10}}}};

Reply:
But that won't work because every shape for data input will be matched with every shape for seqLength, and in the case you described it would be 
{7, 15, 15, 10, 3} and {7} - OK
{7, 15, 15, 10, 3} and {10} - NOT OK
{10, 4, 100, 90, 5} and {7} - NOT OK
{10, 4, 100, 90, 5} and {10} - OK

Reply:
done